controllers/managedocs_controller.go

```diff
@@ -1726,6 +1748,27 @@ func (r *ManagedOCSReconciler) reconcileEgressFirewall() error {
 			)
 		}
 
+		if r.DeploymentType == consumerDeploymentType {
+
+			storageProviderEndpointURL := r.addonParams[storageProviderEndpointKey]
+			storageProviderEndpointHost := strings.Split(storageProviderEndpointURL, ":")[0]
```

Comment:
This might panic if the split is yielding an empty array in the case of a malformed URL.

Reply:
Added a condition to return a error if the split is yielding an empty string